src/classes/ADV_Adapter.cls

```diff
@@ -0,0 +1,126 @@
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2017
+* @group TDTM
+* @group-content ../../ApexDocContent/TDTM.htm
+* @description Provides an adapter for Advancement package to be able to inject customizations into NPSP and provide API to inject to other project.
+**/
+
+global virtual with sharing class ADV_Adapter {
+    /*******************************************************************************************************
+    * @description To be called by the NPSP TDTM trigger dispatcher. Advancement will disable some NPSP trigger
+    * handlers records (since they conflict with HEDA version) and return the rest.
+    * @param listTH The list of Trigger handler record for further processing.
+    * @return List<Trigger_Handler__c> The list of Trigger Handler record for TDTM to use.
+    ********************************************************************************************************/
+    global virtual List<Trigger_Handler__c> getDefaultTdtmConfigNpspRecords(List<Trigger_Handler__c> listTH) {
+        if (!ADV_Info.useAdv()) {
+            throw new AdapterException(Label.adapterException);
+        }
+
+        return TDTM_Config_API.getDefaultRecords();
+    }
+
+    /*******************************************************************************************************
+    * @description Since Advancement will disable some NPSP trigger handler class, we need to fill in gap
+    * in HEDA. Expose this API for Advancement to inject the rollup functionality into HEDA code base
+    * @param listAccountId The list of Account (Id's) that need updating.
+    * @return void
+    ********************************************************************************************************/
+    global void rollupAccounts(List<Id> listAccountId) {
+        if (!ADV_Info.useAdv()) {
+            throw new AdapterException(Label.adapterException);
+        }
+
+        ACCT_IndividualAccounts_TDTM.rollupAccounts(listAccountId);
```

Comment:
@weizsforg This won't work with Customizable Rollups. There currently isn't a single method you can call that will execute the appropriate rollup engine for an Id, but after Customizable Rollups is merged into Master we can work with you allow this method to work no matter which engine is active.

Reply:
@force2b Thanks for the information! Yes, we might need to revisit it. Another thing to mention here is that this method is not a general purpose rollup method that we will use it Gift Processing. We used for a specific purpose.  Basically, we want to achieve the same feature as 
NPSP -> ACCT_IndividualAccounts_TDTM  -> rollupAccounts(list<Id> listAccountId) method. 

So, if the there is no change in NPSP ->  ACCT_IndividualAccounts_TDTM  -> rollupAccounts, then we might not need to change this one.